source/common/router/rds_impl.cc

```diff
@@ -271,6 +309,20 @@ void RdsRouteConfigProviderImpl::validateConfig(
   ConfigImpl validation_config(config, factory_context_, validator_, false);
 }
 
+// Schedules a VHDS request on the main thread and queues up the callback to use when the VHDS
+// response has been propagated to the worker thread that was the request origin.
+void RdsRouteConfigProviderImpl::requestVirtualHostsUpdate(
+    const std::string& for_domain, Event::Dispatcher& thread_local_dispatcher,
+    Http::RouteConfigUpdatedCallbackSharedPtr route_config_updated_cb) {
```

Comment:
I think we can eliminate another concern of if you do Http::RouteConfigUpdatedCallback`Weak`Ptr here

Reply:
Good catch, done.

Reply:
done